src/qibolab/instruments/qm/config/devices.py

```diff
@@ -0,0 +1,84 @@
+from dataclasses import dataclass, field
+
+from qibolab.components.configs import OscillatorConfig
+
+from ..components import OpxOutputConfig, QmAcquisitionConfig
+
+__all__ = [
+    "OctaveOutput",
+    "OctaveInput",
+    "Controller",
+    "Octave",
+]
+
+
+DEFAULT_INPUTS = {"1": {}, "2": {}}
+"""Default controller config section.
+
+Inputs are always registered to avoid issues with automatic mixer
+calibration when using Octaves.
+"""
+
+
+@dataclass(frozen=True)
+class AnalogInput:
+    offset: float = 0.0
+    gain_db: int = 0
+
+    @classmethod
+    def from_config(cls, config: QmAcquisitionConfig):
+        return cls(
+            offset=config.offset,
+            gain_db=config.gain,
+        )
+
+
+@dataclass(frozen=True)
+class OctaveOutput:
+    LO_frequency: int
+    gain: int = 0
+    LO_source: str = "internal"
+    output_mode: str = "triggered"
+
+    @classmethod
+    def from_config(cls, config: OscillatorConfig):
+        return cls(
+            LO_frequency=config.frequency,
+            gain=config.power,
+        )
+
+
+@dataclass(frozen=True)
+class OctaveInput:
+    LO_frequency: int
+    LO_source: str = "internal"
+    IF_mode_I: str = "direct"
+    IF_mode_Q: str = "direct"
+
+
+@dataclass
+class Controller:
+    analog_outputs: dict[str, dict[str, OpxOutputConfig]] = field(default_factory=dict)
+    digital_outputs: dict[str, dict[str, dict]] = field(default_factory=dict)
+    analog_inputs: dict[str, dict[str, AnalogInput]] = field(
+        default_factory=lambda: dict(DEFAULT_INPUTS)
+    )
+
+    def add_octave_output(self, port: int):
+        # TODO: Add offset here?
+        self.analog_outputs[str(2 * port - 1)] = OpxOutputConfig()
```

Comment:
I believe there is no port 0 for QM. Output ports are 1-10 for OPX+ and 1-5 for Octaves. If someone uses `port = 0` here, it will result to a `"0"` and a `"-1"` somewhere in the config which I hope (but have not tested) that it will result to an error from the qm-qua side, since they are validating the config. 

Otherwise, we should raise the errors ourselves, but as long as they are doing it, I would save us the additional lines of code.

Reply:
Also the hardcoded `2 * port - 1`, `2 * port` is due to the way OPX+ ports are connected to Octaves. In principle, you can do screw this up changing these cables, since they are external (not inside the instruments). However, I believe this kind of connectivity is the one suggested by QM and also the most sensible from cable organization point of view, so I decided to also hardcode it here, instead of complicating the driver more to support something that is highly unlikely to be useful.

Reply:
Ok, here I accept your explanation. In any case, if someone with a QM cluster would like to use Qibolab, but they cabled the connections differently, they should be capable enough to get in touch with us (and it wouldn't take that much to fix it).